### PR TITLE
Add resources and cleanup

### PR DESCRIPTION
**Pull request by erinn**
This started out as just doing the needful to get AD integration working in our environment. It sprawled out from there to cleaning up rubocop errors, fixing config options that were marked as required but actually weren't per smb.conf.

---
This change is [Reviewable](https://reviewable.io/reviews/sous-chefs/samba/78)

### CODE REVIEW

**Comment by erinn** [comment]
I'll say this real quick about the checks I cleaned up many many rubocop errors, but not all.

**Comment by damacus** [comment]
Nice one thanks! 

I'll take a look at this one as soon as I can!

**Comment by damacus** [comment]
Reviewed 1 of 7 files at r1, 8 of 9 files at r4.
Review status: 9 of 11 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[Rakefile, line 19 at r4](https://reviewable.io:443/reviews/sous-chefs/samba/78#-KsLEXrdZ2RrPWfxsuEu:-KsLEXrdZ2RrPWfxsuEv:b-dptv90) ([raw file](https://github.com/sous-chefs/samba/blob/390f280aa63d544030cde85aded06820f7e02d1f/Rakefile#L19)):*
> ```Ruby
> 
>     desc 'Run Chef style checks'
>     FoodCritic::Rake::LintTask.new(:chef) do |t|
> ```

We can probably get rid of this file entirely

---

*[resources/share.rb, line 21 at r4](https://reviewable.io:443/reviews/sous-chefs/samba/78#-KsLEzXwySTtZqQ4b3oz:-KsLEzXwySTtZqQ4b3p-:bw1wjuw) ([raw file](https://github.com/sous-chefs/samba/blob/390f280aa63d544030cde85aded06820f7e02d1f/resources/share.rb#L21)):*
> ```Ruby
> property :share_name, String, name_property: true
> property :comment, String
> property :path, String, required: false
> ```

Does it actually work without requiring the path? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sous-chefs/samba/78)*

**Comment by damacus** [comment]
Reviewed 1 of 1 files at r3, 1 of 9 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sous-chefs/samba/78)*

**Comment by damacus** [comment]
Looks good. Do you think there's any way of testing this setup? I take it you've tested this locally at least, but it'd be nice not to break future things

File: resources/server.rb [hunk 1]
```diff
@@ -62,6 +79,12 @@
...
       variables(
+        idmap_config: new_resource.idmap_config,
+        winbind_separator: new_resource.winbind_separator,
+        kerberos_method: new_resource.kerberos_method,
+        encrypt_passwords: new_resource.encrypt_passwords,
+        password_server: new_resource.password_server,
+        realm: new_resource.realm,
         workgroup: new_resource.workgroup,
         server_string: new_resource.server_string,
         security: new_resource.security,
```
Comment (damacus) [inline resources/server.rb]: Can you remove this duplicate key please? 👍

File: resources/server.rb [hunk 2]
```diff
@@ -71,7 +94,9 @@
         load_printers: new_resource.load_printers,
         passdb_backend: new_resource.passdb_backend,
         dns_proxy: new_resource.dns_proxy,
-        samba_options: new_resource.options
+        samba_options: new_resource.options,
+        security: new_resource.security,
+        log_level: new_resource.log_level
```
Comment (damacus) [inline resources/server.rb]: Whack a comma on the end of this and I think linting will pass.

**Comment by erinn** [comment]
*[resources/share.rb, line 21 at r4](https://reviewable.io:443/reviews/sous-chefs/samba/78#-KsLEzXwySTtZqQ4b3oz:-KsLYlfHdteX3obf2Bcf:br34jf1) ([raw file](https://github.com/sous-chefs/samba/blob/390f280aa63d544030cde85aded06820f7e02d1f/resources/share.rb#L21)):*
<details><summary><i>Previously, damacus (Dan Webb) wrote…</i></summary><blockquote>

Does it actually work without requiring the path? 
</blockquote></details>

Yes, this is geared towards the special case of exporting homes via samba. The [homes] section is special in and of itself and it does not require the path, not a lot of other things. ·   If no path was given, the path is set to the user's home directory. So you basically end up fighting the config to get what homes does naturally.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sous-chefs/samba/78)*

**Comment by erinn** [comment]
I think a lot more work is needed to flesh out robust testing for this. That is, to be honest, a bit beyond my scope right now. Not just aiming for getting it working, trying to do a bit better than that, but I can't make it robust enough as I would like.

Yes it works, but it works in a very limited circumstance where your systems is already joined to the AD (assuming we are talk about the ADS part). If you are not in that circumstance, a net ads join would be necessary to prepare the sytstem. My changes get us a bit close, but not all the way to a fully functional encompassing samba cookbook.

As well I have another PR to open to add chefspec matchers (no longer necessary in 7.1.0, but a lot of folks are not using 7.1.0 yet).

---

Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sous-chefs/samba/78#-:-KsL_CYoOYSiVfBShuBS:b-mx58k5)*

**Comment by damacus** [comment]
Thanks!

This current changeset looks good, so I'm going to merge it. 

I'll tidy up the remaining cookstyle errors. In future, I'll not merge until the tests are fully green on Travis (but this one is a minor fix).

I'l release this once we've got some tests, preferably inspec tests, but I'll also accept ChefSpec as a fall back. 

With regards the ChefSpec matchers. I/We are only supporting the latest version of ChefDK and as they're not needed in that, I'd push back and encourage the testing user to update their ChefDK firstly. Unless this is a huge problem. 

- There are a ton of fixes in the latest ChefDK
- Travis only runs the latest ChefDK

**Pull request merged**